readme.md

```diff
@@ -51,16 +59,57 @@ To have same environment as the build script for developing purpose, developer c
 nix develop '.#gcd.<attr>'
 ```
 
-For example, to modify the GCD sources, developer can run:
+For example, to modify the GCD sources, developer might run:
 
 ```bash
 nix develop '.#gcd.gcd-compiled'
 ```
 
 The above command will provide a new bash shell with `mill`, `circt`, `chisel`... dependencies set up.
 
+Certain attributes support direct execution via Nix, allowing arguments to be passed using `--`:
+
+```bash
+nix run '.#gcd.<attr>'
+```
+
+For example, we use elaborator to generate configs for the design. To generate the config for GCDTestbench, developer can run:
+
+```bash
+nix run '.#gcd.gcd-compiled.elaborator' -- config --width 16 --useAsyncReset false
+```
+
+A JSON file named `GCDMain.json` will be generated in the working directory.
+
+As another example, we can run a VCS simulation with waveform trace by:
```

Comment:
Maybe add env requirement for VCS:
https://github.com/chipsalliance/chisel-nix/blob/31bab967b8233a1c8681a6437683a7d02e5f50f8/templates/chisel/nix/overlay.nix#L19-L22